### PR TITLE
add toolchains for armada38x

### PR DESCRIPTION
**Pull request by cytec**
see: https://github.com/SynoCommunity/spksrc/issues/2118

### CODE REVIEW

File: kernel/syno-armada38x-5.2/Makefile [hunk 1]
```diff
@@ -0,0 +1,21 @@
+ARCH ?= armada38x
+TCVERSION ?= 5.2
+PKG_NAME = armada38x
+PKG_BRANCH = 5644
+PKG_ARCH = armada38x
```
Comment (Dr-Bean) [inline kernel/syno-armada38x-5.2/Makefile]: Probably makes sense to use `$(PKG_NAME)` here. We already repeat it more than I'd like ;)

File: kernel/syno-armada38x-5.2/Makefile [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+HOMEPAGE = http://www.synology.com/
+
+PRE_COMPILE_TARGET = myPreCompile
+
+include ../../mk/spksrc.kernel-modules.mk
+
+PHONY: myPreCompile
+myPreCompile:
+	cd $(KERNEL_DIR)/arch/arm/ && rm -rf plat-armada && ln -s plat-armada-v2 plat-armada
```
Comment (Dr-Bean) [inline kernel/syno-armada38x-5.2/Makefile]: As far as I'm aware, this isn't needed for Armada38x. It was broken for Armada375 only (according to Synology anyway). In other words, no need for the override :)

File: toolchains/syno-armada38x-5.2/Makefile [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+TC_ARCH = armada38x
+TC_VERS = 5.2
+TC_FIRMWARE = 5.2-5644
+
+TC_DIST_NAME = armada38x-gcc493_glibc220_hard-GPL.txz
+
+TC_EXT = txz
+TC_DIST_SITE = http://sourceforge.net/projects/dsgpl/files/DSM%205.2%20Tool%20Chains/Marvell%20Armada%2038x%20Linux%203.10.35
+
+TC_BASE_DIR = arm-marvell-linux-gnueabi
```
Comment (Dr-Bean) [inline toolchains/syno-armada38x-5.2/Makefile]: Value should be `arm-unknown-linux-gnueabi` for the next three lines. This follows the dirnames you see after extracting.

File: toolchains/syno-armada38x-5.2/Makefile [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+TC_PREFIX = arm-marvell-linux-gnueabi
+TC_TARGET = arm-marvell-linux-gnueabi
+
+include ../../mk/spksrc.tc.mk
```
Comment (Dr-Bean) [inline toolchains/syno-armada38x-5.2/Makefile]: I believe we'll need a `FIX_TARGET` override in here as well, although I could be wrong.

When not running as `root`, do a `make clean`. If you get a permission denied error, you can fix permissions as follows:

```
FIX_TARGET = myFix

include ../../mk/spksrc.tc.mk

.PHONY: myFix
myFix:
	chmod -R u+w $(WORK_DIR)
```

**Comment by Dr-Bean** [comment]
Thanks :)

I can't actually test if the TC/kernel is working, due to the fact that a 64bit host is needed, and I don't have one at hand, but this is a good start.

**Comment by cytec** [comment]
@Dr-Bean there you go ;) haven't tested because i also don't have a 64bit host available

**Comment by Dr-Bean** [comment]
For 64b hosts, the following flags are necessary as well:
```
+TC_CFLAGS = -I$(WORK_DIR)/$(TC_BASE_DIR)/$(TC_BASE_DIR)/sysroot/usr/include
+TC_CPPFLAGS = -I$(WORK_DIR)/$(TC_BASE_DIR)/$(TC_BASE_DIR)/sysroot/usr/include
+TC_CXXFLAGS = -I$(WORK_DIR)/$(TC_BASE_DIR)/$(TC_BASE_DIR)/sysroot/usr/include
+TC_LDFLAGS = -L$(WORK_DIR)/$(TC_BASE_DIR)/$(TC_BASE_DIR)/sysroot/usr/lib
```
Besides that, `myFix` won't work: Makefiles use tabs. I guess a c/p issue in the other comment:
```
 .PHONY: myFix
 myFix:
-    chmod -R u+w $(WORK_DIR)
+	chmod -R u+w $(WORK_DIR)
```

**Comment by gizmocuz** [comment]
Please correct the digist files, both digist files can not contain the same

**Comment by cytec** [comment]
@Dr-Bean here you go

**Comment by gizmocuz** [comment]
Thanks

**Comment by Dr-Bean** [comment]
Digests still need work, can be done when merged.

**Comment by Dr-Bean** [comment]
Additional changes via 7741d64cf59a1f6a57aaa03898dca98c92efa11d (digest) amd f0156950dd6cecb0b76f4a7dac1e4a4d64209346 (package updates)

**Pull request merged**